include/LinkedModelGroups.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * LinkedModelGroups.h - base classes for groups of linked models
+ *
+ * Copyright (c) 2019-2019 Johannes Lorenz <j.git$$$lorenz-ho.me, $$$=@>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LINKEDMODELGROUPS_H
+#define LINKEDMODELGROUPS_H
+
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include "Model.h"
+
+
+/**
+	@file LinkedModelGroups.h
+	See Lv2ControlBase.h and Lv2Proc.h for example usage
+*/
+
+
+/**
+	Base class for a group of linked models
+
+	See the LinkedModelGroup class for explenations
+
+	Features:
+	* Models are stored by their QObject::objectName
+	* Models are linked automatically
+*/
+class LinkedModelGroup : public Model
+{
+public:
+	/*
+		Initialization
+	*/
+	//! @param parent model of the LinkedModelGroups class
+	LinkedModelGroup(Model* parent) : Model(parent) {}
+
+	/*
+		Linking (initially only)
+	*/
+	void linkControls(LinkedModelGroup *other);
+
+	/*
+		Models
+	*/
+	struct ModelInfo
+	{
+		QString m_name;
+		class AutomatableModel* m_model;
+		ModelInfo() { /* hopefully no one will use this */ } // TODO: remove?
```

Comment:
```suggestion
		ModelInfo() = delete;
```

Reply:
I swear this did not work when I had another container (`std::vector` or `QVector`) in the past, but apparently, it now works with `std::map`. Thanks, I'll fix it.